server/src/main/java/org/opensearch/extensions/rest/ExtensionRestRequest.java

```diff
@@ -291,6 +293,9 @@ public final XContentParser contentParser(NamedXContentRegistry xContentRegistry
         if (!hasContent() || getXContentType() == null) {
             throw new OpenSearchParseException("There is no request body or the ContentType is invalid.");
         }
+        if (!hasContent() || getRequestIssuerIdentity() == null) {
+            throw new OpenSearchParseException("There is no request body or the requester identity is invalid.");
+        }
```

Comment:
This doesn't seem correct, the extension might not be allowed to see the request issuer's identity, I don't think this should thrown an error.

Reply:
You mentioned wanting to throw an error if no token was present. This is that functionality since we handle it as a parsed content and the token is called the issuer identity. This was the existing convention, so I kept it. The idea is that while an extensions may not be able to see the actual issuer but there should be something present whether that is anonymized or not.